pkg/asset/agent/manifests/nmstateconfig.go

```diff
@@ -336,3 +339,33 @@ func buildMacInterfaceMap(nmStateConfig aiv1beta1.NMStateConfig) models.MacInter
 	}
 	return macInterfaceMap
 }
+
+func checkHostCount(installConfig *types.InstallConfig, agentConfig *agentconfig.AgentConfig) {
+	numRequiredMasters, numRequiredWorkers := agent.GetReplicaCount(installConfig)
+
+	// Check that the number of replicas matches the configured hosts
+	numMasters := int64(0)
+	numWorkers := int64(0)
+	for _, host := range agentConfig.Config.Hosts {
+		switch host.Role {
+		case "master":
+			numMasters++
+		case "worker":
+			numWorkers++
+		case "":
+			// If not defined, the roles will be matched to replicas
+			if numMasters < numRequiredMasters {
+				numMasters++
+			} else if numWorkers < numRequiredWorkers {
```

Comment:
This needs to be unconditional if we want to catch the case where there are _more_ hosts defined than replicas.

Reply:
Good point. As this PR is auto cherry-pick of https://github.com/openshift/installer/pull/7059 I'll create a new master PR/bug and fix it there, then cherry-pick that.